### PR TITLE
Add support for SymInt start values in slice_scatter_decomposition

### PR DESCRIPTION
**Pull request by shahe-nv**
# Description

Add support for SymInt start values in slice_scatter_decomposition.

This fixes a crash issue when I run with ZoomASR.

Fixes # (issue): None

## Type of change

- Bug fix (non-breaking change which fixes an issue)

# Checklist:

- [ ] My code follows the style guidelines of this project (You can use the linters)
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas and hacks
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests to verify my fix or my feature
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have added the relevant labels to my PR in so that relevant reviewers are notified

### CODE REVIEW

File: py/torch_tensorrt/dynamo/lowering/_decompositions.py [hunk 1]
```diff
@@ -197,8 +197,13 @@ def slice_scatter_decomposition(
...
-    start = 0 if start is None else start  # Ensure start is int
-    start = get_positive_dim(start, input_tensor.shape[dim])
+    if start is None:
+        start = 0
+    elif isinstance(start, int):
+        start = get_positive_dim(start, dim_size)
+    elif isinstance(start, torch.SymInt):
+        if start < 0:
+            start = start + dim_size
     if end is None:  # Ensure end is int
```
Comment (apbose) [inline py/torch_tensorrt/dynamo/lowering/_decompositions.py:207]: Hi can you have a testcase for the above?

Comment (shahe-nv) [reply py/torch_tensorrt/dynamo/lowering/_decompositions.py:207]: Hi, I tested with my ZoomASR code. Without this, it will crash. After adding this it will run.

Comment (apbose) [reply py/torch_tensorrt/dynamo/lowering/_decompositions.py:207]: Thanks for the info! Generally for any change in the codebase, we recommend adding a corresponding test case to capture the corner case for future CI runs.

Comment (shahe-nv) [reply py/torch_tensorrt/dynamo/lowering/_decompositions.py:207]: Hi, do you know where I need to add the test? Do you have examples?

File: py/torch_tensorrt/dynamo/lowering/_decompositions.py [hunk 2]
```diff
@@ -197,8 +197,13 @@ def slice_scatter_decomposition(
...
 
-    start = 0 if start is None else start  # Ensure start is int
-    start = get_positive_dim(start, input_tensor.shape[dim])
+    if start is None:
+        start = 0
+    elif isinstance(start, int):
+        start = get_positive_dim(start, dim_size)
+    elif isinstance(start, torch.SymInt):
+        if start < 0:
+            start = start + dim_size
```
Comment (zewenli98) [inline py/torch_tensorrt/dynamo/lowering/_decompositions.py:206]: According to [the aten schema](https://github.com/pytorch/pytorch/blob/5bcf6257308559061d638a615da7c47790a8f8e4/aten/src/ATen/native/native_functions.yaml#L5717), all of `start`, `end`, and `step` could be SymInt, not just `start`. Can you fix all of them btw? I think a more general way is to support SymInt in `get_positive_dim()` like:
```
def get_positive_dim(
    dim: Union[int, SymInt, Sequence[int], Sequence[SymInt]], dim_size: int
):
```

File: py/torch_tensorrt/dynamo/lowering/_decompositions.py
Comment (zewenli98) [inline py/torch_tensorrt/dynamo/lowering/_decompositions.py:195]: the types should be changed accordingly, and step=1 by default according to the schema

File: py/torch_tensorrt/dynamo/lowering/_decompositions.py (hunk 1 above)
Comment (zewenli98) [reply py/torch_tensorrt/dynamo/lowering/_decompositions.py:207]: @shahe-nv I think @apbose meant you could add unit tests in `tests/py/dynamo/lowering/test_decompositions.py`

File: py/torch_tensorrt/dynamo/lowering/_decompositions.py
Comment (zewenli98) [inline py/torch_tensorrt/dynamo/lowering/_decompositions.py:213]: not needed if step=1 by default

Comment (zewenli98) [inline py/torch_tensorrt/dynamo/lowering/_decompositions.py:216]: what if they are symint?

Comment (zewenli98) [inline py/torch_tensorrt/dynamo/lowering/_decompositions.py:222]: shouldn't they be put in the front of the function?

**Comment by zewenli98** [comment]
close in favor of https://github.com/pytorch/TensorRT/pull/4236
@shahe-nv Can you try if the new PR works for ZoomASR?
